Dijkstra's Single-Source Shortest Path/Dijkstra.cpp

```diff
@@ -0,0 +1,70 @@
+#include <iostream> 
+#include <limits.h> 
+
+// Define Number of vertices in the graph 
+#define V 7
+
+using namespace std; 
+// function to find the vertex with minimum distance value, from the set of vertices not yet included in shortest path tree 
+int minDistance(int dist[], bool sptSet[]) 
+{ 
+    int min = INT_MAX, min_index; 
+  
+    for (int v = 0; v < V; v++) 
+    {    
+	if (sptSet[v] == false && dist[v] <= min){ 
+            min = dist[v], min_index = v;} 
+    }
+    return min_index; 
+} 
+  
+// A utility function to print the constructed distance array 
+int printSolution(int dist[]) 
+{ 
+    cout<<"Vertex \t\t Distance from Source\n"; 
+    for (int i = 0; i < V; i++) 
+    {
+	 cout<<i<<" \t\t "<<dist[i]<<"\n";
+    }	 
+} 
+  
+// Function that implements Dijkstra's single source shortest path algorithm which takes in two parameter one the graph and other the source vertix
+void dijkstra(int graph[V][V], int src) 
```

Comment:
The interface seems to strictly function only for adjacency matrices. Maybe you can create a Graph class or namespace separately and create a getter there to fetch edge-weights so that this method is agnostic of the graph implementation.